explainaboard/metrics/registry.py

```diff
@@ -30,3 +32,12 @@ def get_metric_config_class(name) -> type[MetricConfig]:
         return EaaSMetricConfig
 
     raise ValueError(f'Invalid Metric {name}')
+
+
+def metric_config_from_dict(dikt: dict):
+    type = dikt.pop('cls_name')
+    config_cls = get_metric_config_class(type)
+    field_names = set(f.name for f in dataclasses.fields(config_cls))
+    return config_cls(
+        **{k: config_cls.dict_conv(k, v) for k, v in dikt.items() if k in field_names}
+    )
```

Comment:
It looks implemnting `MetricConfig.from_dict(d)` is better than manually manipulating the members.

Reply:
I agree, although I'd suggest we save this for a future PR.